buildscripts/build_vpx_windows.sh

```diff
@@ -17,45 +17,49 @@
 
 set -euo pipefail
 
+SCRIPT_DIR=$(dirname $(realpath "$0"))
+
+source "$SCRIPT_DIR"/cross_compile_detection.sh
+
 usage()
 {
     echo "Download and build vpx for the windows cross compiling environment"
     echo "Usage: $0 --arch {x86_64|i686}"
 }
 
-ARCH=""
-
-while (( $# > 0 )); do
-    case $1 in
-        --arch) ARCH=$2; shift 2 ;;
-        -h|--help) usage; exit 1 ;;
-        *) echo "Unexpected argument $1"; usage; exit 1;;
-    esac
-done
+parse_arch "$@"
 
-if [ "$ARCH" != "i686" ] && [ "$ARCH" != "x86_64" ]; then
-    echo "Unexpected arch $ARCH"
-    usage
-    exit 1
-fi
-
-"$(dirname "$0")"/download/download_vpx.sh
+"$SCRIPT_DIR"/download/download_vpx.sh
 
 if [ "${ARCH}" == "x86_64" ]; then
     ARCH_FLAGS="-fno-asynchronous-unwind-tables"
-    VPX_ARCH="x86_64-win64-gcc"
+    TARGET="--target=x86_64-win64-gcc"
+    CROSS_ARG="${ARCH}-w64-mingw32-"
 elif [ "${ARCH}" == "i686" ]; then \
     ARCH_FLAGS=""
-    VPX_ARCH="x86-win32-gcc"
+    TARGET="--target=x86-win32-gcc"
+    CROSS_ARG="${ARCH}-w64-mingw32-"
+elif [ "${ARCH}" == "macos" ]; then \
+    ARCH_FLAGS="-fno-asynchronous-unwind-tables"
```

Comment:
Is it really needed for macOS or did you just blindly copy it from the Windows build script?

Reply:
Just to clarify, during the CI refactor https://github.com/qTox/qTox/pull/6413 a comment was dropped explaining why this compiler flag was needed for the Windows build:

```sh
# There is a bug in gcc that breaks avx512 on 64-bit Windows https://gcc.gnu.org/bugzilla/show_bug.cgi?id=54412
# VPX fails to build due to it.
# This is a workaround as suggested in https://stackoverflow.com/questions/43152633
VPX_CFLAGS="-fno-asynchronous-unwind-tables"
```

(This comment is getting restored in https://github.com/qTox/qTox/pull/6458)

What the comment says is that that cflag is in there just to avoid a Windows/MinGW specific bug in gcc.

Having the workaround for a Windows/MinGW specific bug applied to macOS looks rather suspicious, thus my earlier question.

Reply:
Yeah reasonable, realized after reviewing the PR replacing the comment. Will test and likely remove, I did just carry it over from the Windows build here.

macOS build breaks on two of the deps due to some rpath difference, for some reason, so some work left there too, but will come back to this.